test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -0,0 +1,260 @@
+	totalNumberOfPods = service.CreatePod.Numberofpods
+	for _, s := range service.CreatePod.Describe {
+		output := describePodInfo{nameOfPods: s[podName].(string), countOfVirtualNetworksPerPod: s[podVNsCount].(int), nameOfVirtualNetworkPerPod: strings.Split(s[podVirtualNetwork].(string), ","), nameOfInterfacePerPod: strings.Split(s[podInterfaceName].(string), ",")}
+		podData = append(podData, output)
+
+	}
+	return nil
+}
+
+// formAnnotationStringOfPod forms the annotation string, used in the generation of each pod yaml file.
+func (data *SNTestData) formAnnotationStringOfPod(pod int) string {
+	var annotationString = ""
+	for xPodVN := 0; xPodVN < podData[pod].countOfVirtualNetworksPerPod; xPodVN++ {
+		var podNetworkSpec = "{\"name\": \"" + podData[pod].nameOfVirtualNetworkPerPod[xPodVN] + "\" ,\"interface\": \"" + podData[pod].nameOfInterfacePerPod[xPodVN] + "\" , \"type\": \"" + interfaceType + "\"}"
+		if annotationString == "" {
+			annotationString = "[" + podNetworkSpec
+		} else {
+			annotationString = annotationString + "," + podNetworkSpec
+		}
+	}
+	annotationString = annotationString + "]"
+	return annotationString
+}
+
+// creationOfPodOnNode creates the pod for the specific annotations as per the parsed pod information using the NewPodBuilder API
+func (data *SNTestData) creationOfPodOnNode(ns string, nodeName string) error {
+	var err error
+	labelApp := "testsecpod"
+	for xPod := 0; xPod < totalNumberOfPods; xPod++ {
+		err := data.createPodForSecondaryNetwork(ns, nodeName, xPod, labelApp, resNum)
+		if err != nil {
+			return fmt.Errorf("Error in creating pods.., err: %v", err)
+		}
+	}
+	return err
+}
+
+// getSecondaryInterface shows up the secondary interfaces created for the specific pod and extracts the IP address for the same.
+func (data *SNTestData) getSecondaryInterface(targetPod int, targetInterface int) (string, error) {
+
+	cmd := []string{"/bin/sh", "-c", fmt.Sprintf("ip addr show %s | grep \"inet\" | awk '{print $2}' | cut -d/ -f1", podData[targetPod].nameOfInterfacePerPod[targetInterface])}
+
+	stdout, _, err := data.e2eTestData.RunCommandFromPod(nameSpace, podData[targetPod].nameOfPods, ctrName, cmd)
+	stdout = strings.TrimSuffix(stdout, "\n")
+	if stdout == "" {
+		log.Fatalf("Error: Interface %s not found on %s. err: %v", podData[targetPod].nameOfInterfacePerPod[targetInterface], podData[targetPod].nameOfPods, err)
+	}
+	return stdout, nil
+}
+
+// checkSubnet checks if the IP address to be pinged has the same subnet as the pod from which the IP Address is pinged.
+func (data *SNTestData) checkSubnet(sourcePod int, targetPod int, targetInterface int) (bool, error) {
+	f := false
+
+	for podCheckForSubnet := 0; podCheckForSubnet < podData[sourcePod].countOfVirtualNetworksPerPod; podCheckForSubnet++ {
+		if podData[sourcePod].nameOfVirtualNetworkPerPod[podCheckForSubnet] == podData[targetPod].nameOfVirtualNetworkPerPod[targetInterface] {
+			time.Sleep(5 * time.Second)
```

Comment:
we don't use sleeps in tests as much as possible as it leads to flaky tests or tests that take longer to run than they should
we typically use `wait.Poll`